lib/iris/cube.py

```diff
@@ -1632,7 +1638,33 @@ def dtype(self):
 
     @dtype.setter
     def dtype(self, dtype):
-        self._dtype = dtype
+        if dtype != self.dtype:
```

Comment:
I can see the case that you might want to unset the dtype for lazy data to `None`, which you can't at the moment given the current implementation ... I've no argument against that.